server/recceiver/cfstore.py

```diff
@@ -0,0 +1,207 @@
+                                pvStatus='InActive', \
+                                time=oldTime))
+
+    # now pvNames contains a list of pv's new on this host/ioc
+    for pv in pvNames:
+        ch = client.findByArgs([('~name',pv)])
+        if not ch:
+            '''New channel'''
+            channels.append(createChannel(pv, \
+                                          chOwner=owner, \
+                                          hostName=hostName, \
+                                          iocName=iocName, \
+                                          pvStatus='Active', \
+                                          time=time))
+        elif len(ch) == 1:
+            '''update existing channel: exists but with a different hostName and/or iocName'''
+            channels.append(updateChannel(ch[0], \
+                                          owner=owner, \
+                                          hostName=hostName, \
+                                          iocName=iocName, \
+                                          pvStatus='Active', \
+                                          time=time))
+    client.set(channels=channels)
+
+def updateChannel(channel, owner, hostName=None, iocName=None, pvStatus='InActive', time=None):
+    '''
+    Helper to update a channel object so as to not affect the existing properties
+    '''
+    # properties list devoid of hostName and iocName properties
+    if channel[u'properties']:
+        channel[u'properties'] = [property for property in channel[u'properties'] \
+                         if property[u'name'] != 'hostName' \
+                         and property[u'name'] != 'iocName'\
+                         and property[u'name'] != 'pvStatus']
+    else:
+       channel[u'properties'] = []
+    if hostName != None:
+        channel[u'properties'].append({u'name' : 'hostName', u'owner' : owner, u'value': hostName})
+    if iocName != None:
+        channel[u'properties'].append({u'name' : 'iocName', u'owner' : owner, u'value': iocName})
+    if pvStatus:
+        channel[u'properties'].append({u'name' : 'pvStatus', u'owner' : owner, u'value': pvStatus})
+    if time:
+        channel[u'properties'].append({u'name' : 'time', u'owner' : owner, u'value': time})
+    return channel
+
+def createChannel(chName, chOwner, hostName=None, iocName=None, pvStatus='InActive', time=None):
+    '''
+    Helper to create a channel object with the required properties
+    '''
+    ch = {u'name':chName,u'owner':chOwner,u'properties':[]}
+    if hostName != None:
+        ch[u'properties'].append({u'name' : 'hostName', u'owner' : chOwner, u'value': hostName})
+    if iocName != None:
+        ch[u'properties'].append({u'name' : 'iocName', u'owner' : chOwner, u'value': iocName})
+    if pvStatus:
+        ch[u'properties'].append({u'name' : 'pvStatus', u'owner' : chOwner, u'value': pvStatus})
+    if time:
+        ch[u'properties'].append({u'name' : 'time', u'owner' : chOwner, u'value': time})
```

Comment:
The duplicate code in ```updateChannel()``` and ```createChannel()``` should be consolidated.  It could also be simplified using the builtin function ```locals()```

```python
VS = locals()
for V in ('hostName', 'iocName', 'pvStatus', 'time'):
    if V in VS:
        ch[u'properties'].append({u'name' : V, u'owner' : chOwner, u'value': VS[V]})
```